tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -1859,3 +1859,115 @@ def __call__(self, inputs, state, scope=None):
       new_state = core_rnn_cell.LSTMStateTuple(new_c, new_h)
 
       return new_h, new_state
+
+class BasicConvLSTMCell(core_rnn_cell.RNNCell):
+  """Basic Convolutional LSTM recurrent network cell.
+
+  https://arxiv.org/pdf/1506.04214v1.pdf
+  """
+
+  def __init__(self,
+               shape,
+               filter_size,
+               num_features,
+               forget_bias=1.0,
+               activation=math_ops.tanh,
+               reuse=None):
+    """Initialize the basic Conv LSTM cell.
+    Args:
+      shape: int tuple thats the height and width of the cell
+      filter_size: int tuple thats the height and width of the filter
+      num_features: int thats the depth of the cell 
+      forget_bias: float, The bias added to forget gates (see above).
+      input_size: Deprecated and unused.
+      state_is_tuple: If True, accepted and returned states are 2-tuples of
+        the `c_state` and `m_state`.  If False, they are concatenated
+        along the column axis.  The latter behavior will soon be deprecated.
+      activation: Activation function of the inner states.
+    """
+    self._shape = shape 
+    self._filter_size = filter_size
+    self._num_features = num_features 
+    self._forget_bias = forget_bias
+    self._reuse = reuse
+    self._activation = activation
+
+  @property
+  def state_size(self):
+    return core_rnn_cell.LSTMStateTuple(self._shape, self._shape)
+
+  @property
+  def output_size(self):
+    return self._shape
+
+  def zero_state(self, batch_size, dtype):
+    shape = self._shape 
+    num_features = self._num_features
+    zero_c = array_ops.zeros([batch_size, shape[0], shape[1], num_features], dtype=dtype)
+    zero_h = array_ops.zeros([batch_size, shape[0], shape[1], num_features], dtype=dtype)
+    zero_state = core_rnn_cell.LSTMStateTuple(zero_c, zero_h)
+    return zero_state
+
+  def __call__(self, inputs, state, scope=None):
+    """Long short-term memory cell (LSTM)."""
+    with vs.variable_scope(scope or "conv_lstm_cell", reuse=self._reuse):  # "BasicLSTMCell"
+      # Parameters of gates are concatenated into one multiply for efficiency.
+      (c, h) = state
+
+      concat = _conv_linear([inputs, h], self._filter_size, self._num_features * 4, True)
```

Comment:
If `self._num_features` is 1, will the gates be created? I think the minimum should be 4 here, no?